code/game/atoms.dm

```diff
@@ -260,17 +260,27 @@
 			if(reagents.reagent_list.len)
 				if(user.can_see_reagents()) //Show each individual reagent
 					for(var/datum/reagent/R in reagents.reagent_list)
-						to_chat(user, "[R.volume] units of [R.name]")
+						if(R.volume < 1000)
+							to_chat(user, "[R.volume] units of [R.name]")
+						else
+							to_chat(user, "[round(R.volume / 1000, 0.01)] kilounits of [R.name]")

```

Comment:
replace lines 263 - 266 with:

to_chat(user, "[total_volume%1000]?"[round(total_volume / 1000, 0.01)] kilounits":"[total_volume] units" of various reagents")